src/rocksdb.rs

```diff
@@ -276,7 +278,8 @@ impl DB {
     	if cfs.len() != cf_opts.len() {
 			return Err(format!("Mismatching number of CF options"));
 		}
-        let cpath = match CString::new(path.as_bytes()) {
+        let encoded_path = Encoding::ANSI.to_bytes(path);
+        let cpath = match CString::new(encoded_path.unwrap()) {
```

Comment:
(our goal is zero panics in Parity code, so all panickers should be backed by a "proof" of why they won't fail)